### DISCUSSION TITLE
Setting NORDIGEN_ID and KEY in the .fidi.env docker-compose file leads to "The Nordigen API is configured incorrectly"

### DISCUSSION THREAD

**Discussion by axelsimon** [firefly-iii/firefly-iii#6934] [Firefly III] [answered]
I'm a bit baffled. I've tested my ID and key with the curl command suggested on the Nordigen user-secrets page
```
    curl -X POST "https://ob.nordigen.com/api/v2/token/new/" \
        -H 'accept: application/json' \
        -H 'Content-Type: application/json' \
        -d '{
            "secret_id": "$SECRET_ID",
            "secret_key": "$SECRET_KEY"
        }'
```
and it works fine.
But setting the same values as `NORDIGEN_ID` and `NORDIGEN_KEY` in the .fidi.env file for my docker-compose setup results in FIDI telling me that "The Nordigen API is configured incorrectly and cannot be used to import data." and the logs to show the following:
```
fidi_1  | [2023-01-26 11:46:13] local.ERROR: Client error: `POST https://ob.nordigen.com/api/v2/token/new/` resulted in a `401 Unauthorized` response:
fidi_1  | {"summary":"Authentication failed","detail":"No active account found with the given credentials","status_code":401}
fidi_1  |  {"exception":"[object] (GuzzleHttp\\Exception\\ClientException(code: 401): Client error: `POST https://ob.nordigen.com/api/v2/token/new/` resulted in a `401 Unauthorized` response:
fidi_1  | {\"summary\":\"Authentication failed\",\"detail\":\"No active account found with the given credentials\",\"status_code\":401}
```

I've tried with and without double quotes around the values in the env file, no difference.
What am i missing?

**Comment by axelsimon** [comment]
Ok, even weirder, the same issue remains after setting the values back to nothing (`NORDIGEN_ID=` and `NORDIGEN_KEY=`).
They must be cached somewhere (in `firefly_iii_db` volume?) and stopping and starting the container does not clear that?

**Comment by JC5** [comment, marked as answer]
Browse to /flush or press Reauthenticate. The old data may have been cached.

**Reply by axelsimon** [reply]
Sadly, stopping and starting (`up`) the fidi container and loading $fidi_host/flush immediately after container startup still results in the same issue 🫤

**Reply by JC5** [reply]
Then I'm not sure. You can try in an incognito browser?

**Reply by axelsimon** [reply]
aaand that does it… in the meantime i destroyed the container about 10 times and updated all the containers. Ah, silly browser cache. Thanks!
